### ISSUE TITLE
ISO C++ forbids flexible array member ‘data’

### ISSUE THREAD

**Issue by obgm** [obgm/libcoap#187]
In PR #162, commit 31224284 has introduced the following new type in `option.h`:

```c
typedef struct coap_optlist_t {
  struct coap_optlist_t *next;  /**< next entry in the optlist chain */
  uint16_t number;              /**< the option number (no delta coding) */
  size_t length;                /**< the option value length */
  uint8_t data[];               /**< the option data */
} coap_optlist_t;
```

C++ as well as certain C compilers forbid the flexible array member:

```
include/coap/option.h:427:16: warning: ISO C++ forbids flexible array member ‘data’ [-Wpedantic]
   uint8_t data[];               /**< the option data */
```

**Comment by mrdeep1** [comment]
Apologies - this struct was a copy with enhancements from examples/coap_list.h coap_list_t struct.

I will get this fixed

**Comment by mrdeep1** [comment]
Fixed in PR #188
